### PR TITLE
Made ODH cert default cert for Token Auth

### PR DESCRIPTION
**Pull request by Bobbins228**
# Issue link

[RHOAIENG-52](https://issues.redhat.com/browse/RHOAIENG-52), [RHOAIENG-4375](https://issues.redhat.com/browse/RHOAIENG-4375)
# What changes have been made

For `TokenAuthentication` the SDK will use the cert injected into a ODH/RHOAI Notebook by default in the `/etc/pki/tls/custom-certs/ca-bundle.crt` location
# Verification steps

### Required steps for ODH/RHOAI
* ODH > 2.9
* Create a Data Science Workbench
* Ensure that the path `/etc/pki/tls/custom-certs/ca-bundle.crt` exists in the notebook.
### Notebook server ODH
* Clone this repository with `git clone https://github.com/project-codeflare/codeflare-sdk.git`
* Checkout this PR's branch
* Run `poetry build` - install if needed (`pip install poetry`)
* Run `pip install --force-reinstall dist/codeflare_sdk-0.0.0.dev0-py3-none-any.whl` 
* Restart your notebook kernel
## Testing
Authenticate with `TokenAuthentication`.
Set the environment variable `CA_CERT_PATH` to the path to your cert before authenticating 
```
auth = TokenAuthentication(
    token = "XXXXX",
    server = "XXXXX",
    skip_tls=False
)
auth.login()
```
You should receive the message `Authenticated with certificate located at /etc/pki/tls/custom-certs/ca-bundle.crt`
You should be able to run through any demo notebook.
## Checks
- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Comment by dimakis** [comment]
Whats the status of this PR? it looks like the PR with which it depends (kubeflow) on has been merged. 
we need this for disconnected correct?

**Comment by Bobbins228** [comment]
> Whats the status of this PR? it looks like the PR with which it depends (kubeflow) on has been merged. 
> we need this for disconnected correct?

Kubeflow PR is in this is good to merge after a review and yeah pretty sure it's needed for disconnected

File: src/codeflare_sdk/cluster/auth.py [hunk 1]
```diff
@@ -101,12 +101,16 @@ def login(self) -> str:
         """
         global config_path
         global api_client
+        odh_ca_path = "/etc/pki/tls/custom-certs/ca-bundle.crt"
```
Comment (KPostOffice) [inline src/codeflare_sdk/cluster/auth.py:104]: Can we move this into `__init__` and set `self.ca_cert_path` directly? and also allow the default path to be configurable via an environment variable?

**Review by KPostOffice — COMMENTED** [review]
Nice changes, should make things much smoother!

File: src/codeflare_sdk/cluster/auth.py (hunk 1 above)
Comment (Bobbins228) [reply src/codeflare_sdk/cluster/auth.py:104]: Those are some great suggestions I'll make those changes

File: src/codeflare_sdk/cluster/auth.py [hunk 2]
```diff
@@ -106,10 +106,24 @@ def login(self) -> str:
             configuration.api_key_prefix["authorization"] = "Bearer"
             configuration.host = self.server
             configuration.api_key["authorization"] = self.token
-            if self.skip_tls == False and self.ca_cert_path == None:
+            ca_path_env = os.environ.get("CA_CERT_PATH")
```
Comment (KPostOffice) [inline src/codeflare_sdk/cluster/auth.py:109]: os.environ.get("CF_SDK_CA_CERT_PATH", ca_path_env)

os.environ.get allows defaulting which can be used here. We should also prefix env vars WDYT of `CF_SDK_`

Comment (Bobbins228) [reply src/codeflare_sdk/cluster/auth.py:109]: Sounds good to me. I should also update the documentation in this PR too

**Comment by KPostOffice** [comment]
see #544

**Pull request closed without merging**